### PR TITLE
Added consumes usage to HcalHardcodeCalibrations

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This included use of ESGetTokens.

#### PR validation:

runTheMatrix.py jobs using the module still run.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: CalibCalorimetry/HcalPlugins/src/HcalHardcodeCalibrations.cc [hunk 1]
```diff
@@ -193,75 +193,85 @@ HcalHardcodeCalibrations::HcalHardcodeCalibrations(const edm::ParameterSet& iCon
...
+      setWhatProduced(this, &HcalHardcodeCalibrations::produceChannelQuality).setConsumes(topoTokens_[kChannelQuality]);
       findingRecord<HcalChannelQualityRcd>();
     }
     if ((objectName == "ElectronicsMap") || (objectName == "electronicsMap") || all) {
-      setWhatProduced(this, &HcalHardcodeCalibrations::produceElectronicsMap);
+      setWhatProduced(this, &HcalHardcodeCalibrations::produceElectronicsMap).setConsumes(topoTokens_[kElectronicsMap]);
       findingRecord<HcalElectronicsMapRcd>();
     }
     if ((objectName == "ZSThresholds") || (objectName == "zsThresholds") || all) {
       setWhatProduced(this, &HcalHardcodeCalibrations::produceZSThresholds);
```
Comment (Dr15Jones) [inline CalibCalorimetry/HcalPlugins/src/HcalHardcodeCalibrations.cc]: setConsumes is missing from here

**Comment by Dr15Jones** [comment]
please test

**Comment by pohsun** [comment]
+1

**Comment by Dr15Jones** [comment]
This does NOT depend on #28223 so the `requires-external` label is incorrect.

**Comment by fabiocos** [comment]
@Dr15Jones the ```require-external``` addition is a feature related to the request of additional stuff, it does not necessarily mean what we do, not sure whether we have a trivial way out (@smuzaffar FYI). Anyway thanks for the clarification, usually this is enough for a proper code management

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Comment by Dr15Jones** [comment]
The test failures are from #28223  not this pull request.

**Pull request merged**